typegen/src/utils.rs

```diff
@@ -34,7 +34,7 @@ pub fn ensure_unique_type_paths(types: &mut PortableRegistry) {
             let ty_id_b = group[0]; // all types in group are same shape; just check any one of them.
             let ty_b = types.resolve(ty_id_b).expect("ty exists");
             if types_equal_extended_to_params(&ty.ty, ty_b) {
-                group.push(ty_id_b);
+                group.push(ty.id);
```

Comment:
This is it. This caused all the trouble.

Reply:
Good job finding this :)